bd-logger/src/consumer.rs

```diff
@@ -601,9 +629,24 @@ impl CompleteBufferUpload {
     let mut total_logs = 0;
     loop {
       let entry = self.consumer.try_read();
+
       match entry {
         // Log available, add to batch and flush if batch size hit.
         Ok(log) => {
+          if let Ok(log) = root_as_log(&log) {
+            if let Some(lookback_window) = self.lookback_window {
+              if let Some(ts) = log.timestamp() {
```

Comment:
Is it possible for there to ever not have a timestamp or is this an artifact of the fb API? Might be worth a comment.